packages/utils/dashboard/status-card/HealthItem.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import { HealthState } from '@openshift-console/dynamic-plugin-sdk';
+import classNames from 'classnames';
+import { useTranslation } from 'react-i18next';
+import { Button, Popover, PopoverPosition } from '@patternfly/react-core';
+import SecondaryStatus from '../../status/SecondaryStatus';
+import { healthStateMapping, healthStateMessage } from './states';
+
+export type HealthItemProps = {
+  className?: string;
+  title: string;
+  details?: string;
+  state?: HealthState;
+  popupTitle?: string;
+  noIcon?: boolean;
+  icon?: React.ReactNode;
+  maxWidth?: string;
+};
+
+const HealthItemIcon: React.FC<HealthItemIconProps> = ({ state, dataTest }) => (
+  <div data-test={dataTest} className="co-dashboard-icon">
+    {
+      (healthStateMapping[state] || healthStateMapping[HealthState.UNKNOWN])
+        .icon
+    }
+  </div>
+);
+
+// eslint-disable-next-line react/display-name
+const HealthItem: React.FC<HealthItemProps> = React.memo(
+  ({
+    className,
+    state,
+    title,
+    details,
+    popupTitle,
+    noIcon = false,
+    icon,
+    children,
+    maxWidth,
+  }) => {
+    const { t } = useTranslation();
+
+    const detailMessage = details || healthStateMessage(state, t);
+
+    return (
+      <div
+        className={classNames('co-status-card__health-item', className)}
+        data-item-id={`${title}-health-item`}
+      >
+        {state === HealthState.LOADING ? (
+          <div className="skeleton-health">
+            <span className="pf-u-screen-reader">
+              {t('Loading {{title}} status', { title })}
+            </span>
+          </div>
+        ) : (
+          !noIcon &&
+          (icon || (
+            <HealthItemIcon
+              state={state}
+              dataTest={`${title}-health-item-icon`}
```

Comment:
It's always better to use the static name for dataTest, otherwise, the test case will keep breaking when any changes on the title.